src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1015,30 +1153,20 @@ private void ImportStoreIndirect(int token)
         private void ImportStoreIndirect(TypeDesc type)
         {
             StackEntry value = _stack.Pop();
-            StackEntry destinationPointer = _stack.Pop();
-            LLVMTypeRef requestedPointerType = LLVM.PointerType(GetLLVMTypeForTypeDesc(type), 0);
-            LLVMValueRef typedValue = value.LLVMValue;
-            LLVMValueRef typedPointer = destinationPointer.LLVMValue;
+            var destinationPointer = _stack.Pop() as ExpressionEntry;
+            var maybeType = type ?? value.Type;
```

Comment:
I'm not sure I understand what maybeType represents. Can you comment what this is?